core/parseXml.ml

```diff
@@ -1,66 +1,17 @@
-open Lexing
-open SourceCode
+open Parse
 
-(*
-   This module is a copied, pasted and hacked version of parse.ml
-*)
-
-type 'a grammar = (Lexing.lexbuf -> XmlParser.token) -> Lexing.lexbuf -> 'a
-
-class source_code = SourceCode.source_code
-
-(* Read and parse Links source code from the source named `name' via
-   the function `infun'.
-*)
-let read : context:XmlLexer.lexer_context
-        -> ?nlhook:(unit -> unit)
-        -> parse:('intermediate grammar)
-        -> infun:(bytes -> int -> int)
-        -> name:string
-        -> 'result * source_code =
-fun ~context ?nlhook ~parse ~infun ~name ->
-  let code = new source_code in
-  let lexbuf = {(from_function (code#parse_into infun))
-                 with lex_curr_p={pos_fname=name; pos_lnum=1; pos_bol=0; pos_cnum=0}} in
-   try
-      let p = parse (XmlLexer.lexer context ~newline_hook:(Utility.from_option Utility.identity nlhook)) lexbuf in
-        (p, code)
+module XmlLexer : (LexerSig with type token         = XmlParser.token and
+                                 type lexer_context = XmlLexer.lexer_context) =
+  struct
+    type token         = XmlParser.token
+    type lexer_context = XmlLexer.lexer_context
+    type 'a grammar    = (Lexing.lexbuf -> XmlParser.token) -> Lexing.lexbuf ->
+                         'a
```

Comment:
looks a bit weird that `'a` is on its own line.

Reply:
Done